lib/transformer.js

```diff
@@ -0,0 +1,288 @@
+'use strict';
+
+var async = require('async');
+var isString = require('101/is-string');
+var FsDriver = require('./fs-driver');
+var Warning = require('./warning');
+var exists = require('101/exists');
+
+/**
+ * @module fs-transform:Transformer
+ * @author Ryan Sandor Richards
+ */
+module.exports = Transformer;
+
+/**
+ * Class that performs a series of filesystem tranformations given a root
+ * directory and list of rules.
+ * @class
+ * @author Ryan Sandor Richards
+ * @param {String} root Root directory to run the transformations.
+ * @param {String|Array} rules An array of transformations, or a JSON string that
+ *   parses into a stream of transformations.
+ */
+function Transformer(root, rules) {
+  if (isString(rules)) {
+    rules = JSON.parse(rules);
```

Comment:
I think I would like the transformer class to throw the exception, we explicitly handle it down in the `transform` static helper function below. I'll add jsdoc to indicate when it throws.